lib/js/components/Switch/Switch.vue

```diff
@@ -0,0 +1,350 @@
+<template>
+	<div class="switchWrapper">
+		<div
+			class="switch"
+			:class="{
+				'-small': size === SWITCH_SIZE.SMALL,
+				'-medium': size === SWITCH_SIZE.MEDIUM,
+
+				'-rounded': radius === SWITCH_RADIUSES.ROUNDED,
+
+				'-disabled': state === SWITCH_STATE.DISABLED,
+			}"
+		>
+			<div
+				class="switch__item -left"
+				:class="{
+					'-selected': currentSide === SWITCH_SIDE.LEFT,
+				}"
+				:title="labelLeft"
+				@click="onSwitch(SWITCH_SIDE.LEFT)"
+			>
+				<ds-icon
+					v-if="iconLeft"
+					class="switch__icon"
+					:icon="iconLeft"
+					:size="ICON_SIZES.XX_SMALL"
+				/>
+				<div v-if="labelLeft" class="switch__label">
+					{{ labelLeft }}
+				</div>
+			</div>
+			<div
+				class="switch__item -right"
+				:class="{
+					'-selected': currentSide === SWITCH_SIDE.RIGHT,
+				}"
+				:title="labelRight"
+				@click="onSwitch(SWITCH_SIDE.RIGHT)"
+			>
+				<ds-icon
+					v-if="iconRight"
+					class="switch__icon"
+					:icon="iconRight"
+					:size="ICON_SIZES.XX_SMALL"
+				/>
+				<div v-if="labelRight" class="switch__label">
+					{{ labelRight }}
+				</div>
+			</div>
+			<div
+				class="switch__item -selection"
+				:class="{
+					'-left': currentSide === SWITCH_SIDE.LEFT,
+					'-right': currentSide === SWITCH_SIDE.RIGHT,
+				}"
+			>
+				<ds-icon
+					v-if="currentIcon"
+					class="switch__icon"
+					:icon="currentIcon"
+					:size="ICON_SIZES.XX_SMALL"
+				/>
+				<div v-if="currentLabel" class="switch__label">{{ currentLabel }}</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../styles/settings/animations';
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/radiuses';
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/typography/tokens';
+
+$switch-transition: all $default-transition-time ease-out;
+
+.switchWrapper {
```

Comment:
Wy do we need a wrapper?

Reply:
To prevent overflowing options in case just one of them is too long for the parent container. It's an edge case. I get rid of it if you believe this is not going to be the case :+1:

Reply:
Let's leave it :)